tezos_interop/tezos_interop.re

```diff
@@ -748,6 +790,52 @@ module Consensus = {
       ~on_message,
     );
   };
+  let fetch_validators = (~context) => {
+    let Context.{rpc_node, required_confirmations, consensus_contract, _} = context;
+    let micheline_to_validators =
+      fun
+      | Ok(
+          Micheline.Prim(
+            _,
+            Michelson_v1_primitives.D_Pair,
+            [
+              _,
+              Prim(
+                _,
+                D_Pair,
+                [
+                  _,
+                  Prim(_, D_Pair, [_, Prim(_, D_Pair, secret_keys, _)], _),
+                ],
+                _,
+              ),
+            ],
+            _,
+          ),
+        ) => {
+          Ok(
+            List.map(
+              fun
+              | Micheline.String(_, sk) =>
+                switch (Secret.of_string(sk)) {
+                | Some(sk) => sk
+                | None => failwith("sk isn't a valid Secret")
+                }
+              | _ => failwith("Should have been String"),
```

Comment:
This should not be a failwith, I think there is a `fold_left_ok` if you want you can do a make `map_ok`

Reply:
Didn't get you - what's to be done with keys that dont parse?

Reply:
return an Error()

Reply:
We could still use the keys that parse - what about a filter_map?

Reply:
If a single key failed then everything needs to fail.